### ISSUE TITLE
feature(client): Show Client ID somewhere in the Client

### ISSUE THREAD

**Issue by bmanifold** [CloudFire-LLC/cloudfire#6656]
As a Firezone admin, I would like to be able to physically verify that a device I see in the Admin Portal is the same device I have physical access to.  

At a minimum, I would like the UUID that is shown in the Admin Portal on be displayed somewhere on the Client device.  It would be even better if I could see more information about the Client/device in the Admin Portal to allow stronger verification, e.g. MAC addresses of all interfaces, kernel version, serial number, device model number, etc...

**Comment by jamilbk** [comment]
Requirement on iOS if we want to actually show device name is that we need to:

- request an entitlement from Apple
- actually use that / show it somewhere in the app UI (what this issue is)

**Comment by thomaseizinger** [comment]
> * request an entitlement from Apple

Why? We already have the device name to send it to the portal. Does displaying really require its own permission?

**Comment by jamilbk** [comment]
> > * request an entitlement from Apple
> 
> Why? We already have the device name to send it to the portal. Does displaying really require its own permission?

We don't have it. We only get a generic name: `iPhone` when requesting it. This is a privacy feature to prevent apps from fingerprinting users by `Jamil's iPhone` - the default name for example.

https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_device-information_user-assigned-device-name

**Comment by thomaseizinger** [comment]
> > > * request an entitlement from Apple
> > 
> > 
> > Why? We already have the device name to send it to the portal. Does displaying really require its own permission?
> 
> We don't have it. We only get a generic name: `iPhone` when requesting it. This is a privacy feature to prevent apps from fingerprinting users by `Jamil's iPhone` - the default name for example.
> 
> [developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_device-information_user-assigned-device-name](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_device-information_user-assigned-device-name)

We don't necessarily have to show the device name right? We already uniquely identify device by ID so we can show that one instead?

**Comment by jamilbk** [comment]
Yeah, it looks like the portal stores this as `external_id`, so we just need to display this field on both sides to verify the connected client.

I believe `external_id = sha256(firezone_id)` where `firezone_id` is the persisted UUID in the client.

**Comment by jamilbk** [comment]
If we simply go back to reading the hardware serial and sending this unmodified to the portal as `external_id`, admins will be able to verify clients against other systems (MDM, XDR) to verify the client as well.

**Comment by jamilbk** [comment]
refs https://github.com/firezone/firezone/issues/6837
